File: 이건희/level2/기능개발.py
from math import ceil

def solution(progresses, speeds):
    result = []
    cnt = 1
    day = 0
    
    for i in range(len(progresses)):
        num = ceil((100 - progresses[i]) /speeds[i])
        if day < num:
            day = num
            result.append(cnt)
            cnt = 1
        else:
            cnt += 1
            
    return result[1:] + [cnt]

File: 김찬우/프로그래머스 /Lv.2/기능개발.java
import java.util.*;
import java.lang.Math;
class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        
        List<Integer> result = new ArrayList<>();
        
        Queue<Integer> queue = new LinkedList();
        
        for(int i = 0; i<progresses.length; i++){
             queue.offer((int)Math.ceil((100.0-progresses[i])/speeds[i]));
        }

        
        while(!queue.isEmpty()){
            
            int minDays = queue.poll();
            
            int count = 1;
            
            while(!queue.isEmpty() && queue.peek() <= minDays){
                queue.poll();
                count++;
            }
            
            result.add(count);
            
            
        }

        return result;
    }
}
